lib/src/license.dart

```diff
@@ -11,55 +11,36 @@ import 'dart:io';
 import 'package:path/path.dart' as p;
 
 import 'download_utils.dart';
+import 'maintenance.dart';
 import 'model.dart';
 
-Future<List<LicenseFile>> detectLicensesInDir(String baseDir) async {
-  final rootFiles = await Directory(baseDir).list().toList();
-  final licenseCandidates = rootFiles
-      .whereType<File>()
-      .where(_isLicenseFile)
-      .take(5) // only the first 5 files are considered
-      .toList();
-  if (licenseCandidates.isEmpty) return [];
-
-  final licenses = await Future.wait(licenseCandidates.map(
-    (File file) {
-      final relativePath = p.relative(file.path, from: baseDir);
-      return detectLicenseInFile(file, relativePath: relativePath);
-    },
-  ));
-
-  licenses.sort((l1, l2) => Comparable.compare(l1.path, l2.path));
-  return licenses;
+Future<LicenseFile> detectLicensesInDir(String baseDir) async {
+  for (final candidate in licenseFileNames) {
```

Comment:
Historically, the `pub` tool has a allow license files with many different names... `pana` and `pub.dev` needs to continue to support such old packages.

But I'm fine with saying that:
 * if there is a `LICENSE` file, then it is always the license we show, and,
 * if there is no `LICENSE` file, then we reject the package when published.

Note, that for existing packages, the old license extraction logic needs to be preserved, does it not?

Reply:
> the old license extraction logic needs to be preserved

With the new rules we'll lose 4 licenses detected out of 10k+ (latest stable versions):
- `LICENSE.html` (2 instances)
- `LICENSE.GPLv3`
- `LICENSE.MIT`

Supporting and hosting a free-form `.html` is not something we want to do or encourage.

The remaining `LICENSE.<name>` is debatable - do we want to keep supporting them?

Reply:
Btw. the current logic now follows the same selection precedence as we have in https://github.com/dart-lang/pub-dev/pull/3908